compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp

```diff
@@ -472,34 +472,62 @@ static LogicalResult setRootConfigForPadPackPipeline(
 
 static LogicalResult setRootConfigForConvDecomposePipeline(
     mlir::FunctionOpInterface entryPointFn, linalg::LinalgOp linalgOp) {
+  MLIRContext *context = entryPointFn.getContext();
+
   FailureOr<std::array<uint32_t, 3>> maybeInstructionSize =
       getMatmulInstructionSize(linalgOp);
   int64_t OW = 4;
   int64_t OC = 4;
   int64_t IC = 8;
   if (succeeded(maybeInstructionSize)) {
+    // (m, n, k)
     auto instructionSize = maybeInstructionSize.value();
     OW = instructionSize[0];
     OC = instructionSize[1];
     IC = instructionSize[2];
   }
 
+  SmallVector<int64_t> transposePackIndices{0, 1, 2};
+  SmallVector<bool> unpackEmpty{false, false, true};
+
+  // Convolution type specific vectors:
+  SmallVector<SmallVector<int64_t>> innerPerm;
+  SmallVector<SmallVector<int64_t>> outerPerm;
   SmallVector<int64_t> tileSizeLevel0;
   SmallVector<int64_t> tileSizeLevel1;
-  SmallVector<int64_t> tileSizeLevel2;
-  // Note: some of the tiling dimensions are hardcoded for now.
-  if (isa<linalg::Conv2DNhwcHwcfOp>(linalgOp) ||
-      isa<linalg::Conv2DNhwcHwcfQOp>(linalgOp)) {
-    // conv_2d_nhwc_hwcf tiling dims: [N, OH, OW, OC, KH, KW, IC].
-    tileSizeLevel0 = {0, 4, OW, OC, 0, 0, 0};
+  SmallVector<int64_t> packingSizes;
+
+  // [N, OH, OW, OC, KH, KW, IC].
+  if (isa<linalg::Conv2DNhwcHwcfQOp>(linalgOp) ||
+      isa<linalg::Conv2DNhwcHwcfOp>(linalgOp)) {
+    innerPerm = {{}, {{1, 0}}, {}};
+    outerPerm = {{0, 1, 3, 2}, {}, {0, 1, 2, 3}};
+    packingSizes = {0, 0, 0, OC, 0, 0, IC};
     tileSizeLevel1 = {1, 1, OW, OC, 0, 0, 0};
-    tileSizeLevel2 = {0, 0, 0, 0, 1, 1, IC};
-  } else if (isa<linalg::Conv2DNchwFchwOp>(linalgOp)) {
-    // conv_2d_nchw_fchw tiling dims: [N, OC, OH, OW, IC, KH, KW].
-    tileSizeLevel0 = {0, OC, 4, OW, 0, 0, 0};
+  }
+
+  // [N, OC, OH, OW, IC, KH, KW]
+  else if (isa<linalg::Conv2DNchwFchwOp>(linalgOp)) {
+    // Packing mplemented below:
+    // - Pack without any dimension transposition.
+    //   With PR https://github.com/nod-ai/iree-amd-aie/pull/787 sizes can be
+    //   {0, 0, 0, 0, 0, 0, 0}, 3 true identity pack operation. Without, we need
+    //   to trick the compiler into not eliding pack operations, hence the 1's.
+    //
+    // Longer-term plan:
+    // - Pack the input image the same way as conv_2d_nhwc_hwcf
+    // - Do not pack/transpose the kernel.
+    // - Perform tranpose of kernel on core.
+    // Currently this is not possible because of limitations on the pack op.
+    // Currently, scalar code is generated (peano now supports this).
+    packingSizes = {0, 1, 0, 0, 1, 0, 0};
+    outerPerm = {{0, 1, 2, 3}, {0, 1, 2, 3}, {0, 1, 2, 3}};
```

Comment:
I think the regular outerPerm order is {0, 1, 2, 3}? Do you have to specify this?

Reply:
Somewhere in the compilation flow I was getting an error that either innerPerm or outerPerm must be specified. I can't remember where, but I'll take a look at this again